clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -24,18 +24,23 @@
 public final class RecordMetadata {
 
     private final long offset;
+    // The timestamp of the message.
+    // If LogAppendTime is used for the topic, the timestamp will be a timestamp returned by broker.
+    // If CreateTime is used for the topic, the timestamp is the timestamp in the corresponding ProducerRecord.
```

Comment:
If CreateTime is used, we are returning user provided timestamp if it exists or the time when the record was handed to the producer.